index.js

```diff
@@ -164,7 +164,9 @@ class Rewt {
   /**
    * Clears the cached secret and clearing timeout.
    */
-  _clearCachedSecret() {
+  _clearCachedSecret(secret) {
+    if (!!secret && secret === this._cachedSecret) return;
```

Comment:
Shouldn't this condition be `secret !== this._cachedSecret`? We know that `secret` is invalid, and we want to clear the cached secret if it's the same.